src/components/ModalWindow/ModalWindow.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+
+import styles from "./index.module.scss";
+
+import styled from "styled-components";
+
+import { colors } from "../styles/colors.js";
+
+import { MetModalWindowProps } from "./ModalWindow.interface";
+import MetCircleIconBtn, { SizeCircleIconBtn } from "../CircleIconBtn";
+import MetIcon, { Icons } from "../Icon";
+import MetRectangleTextBtn, {
+  SizeRectangleTextBtn,
+  TypesRectangleTextBtn,
+} from "../RectangleTextBtn";
+import { useEffect } from "react";
+
+const classNames = require("classnames");
+
+export const MetModalWindow: React.FC<MetModalWindowProps> = ({
+  style,
+  className = "",
+  fontClass = styles.subtitle3,
+  icon = Icons.errorOutline,
+  title,
+  text,
+  acceptText,
+  cancelText,
+  acceptColor = colors.red200,
+  cancelColor = "transparent",
+  iconColor = colors.red200,
+  isDisplayed = false,
+  acceptOnClick,
+  cancelOnClick,
+  onClose,
+}) => {
+  useEffect(() => {
+    document.addEventListener("keydown", escFunction, false);
+    return () => {
+      document.removeEventListener("keydown", escFunction, false);
+    };
+  }, []);
+  const escFunction = (event) => {
+    if (event.keyCode === 27) {
+      onClose();
+    }
+  };
+  const acceptOnClickHandler = (event) => {
+    acceptOnClick && acceptOnClick(event);
+    onClose();
```

Comment:
Why you do not checked for availability of onClose() like acceptOnClick?

onClose && onClose(event);